src/OrchardCore.Modules/OrchardCore.Flows/Drivers/FlowPartDisplayDriver.cs

```diff
@@ -47,11 +59,29 @@ public override IDisplayResult Display(FlowPart flowPart, BuildPartDisplayContex
 
         public override IDisplayResult Edit(FlowPart flowPart, BuildPartEditorContext context)
         {
-            return Initialize<FlowPartEditViewModel>(GetEditorShapeType(context), m =>
+            return Initialize<FlowPartEditViewModel>(GetEditorShapeType(context), async model =>  
             {
-                m.FlowPart = flowPart;
-                m.Updater = context.Updater;
-                m.ContainedContentTypeDefinitions = GetContainedContentTypes(context.TypePartDefinition);
+                var containedContentTypes = GetContainedContentTypes(context.TypePartDefinition);
+                var notify = false;
+
+                foreach (var widget in flowPart.Widgets)
+                {
+                    if (!containedContentTypes.Any(c => c.Name == widget.ContentType))
```

Comment:
would it be better if here we constructed a new list of widgets that do have a content type, and pass that to the view model?

Savings us the need to make any changes and checks in the View

Reply:
I agree.

Reply:
Just this to do before merging I think?

Reply:
We could merge and optimize the code later on too. Should end up with the same result.

Reply:
Done